backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -441,4 +445,72 @@ public RequestStatus updateProjectVulnerabilityRating(ProjectVulnerabilityRating
         }
         return projects;
     }
+
+    public List<Project> fillClearingStateSummaryIncludingSubprojects(List<Project> projects, User user) {
+        // load all projects in a single db request first for performance reasons
+        final Map<String, Project> allProjectsIdMap = ThriftUtils.getIdMap(repository.getAll());
```

Comment:
So, here's the big problem I see with this approach.
You've chopped the request for clearing status into many requests and for each of these requests, you load the complete projects repository. That means, to display the projects page for one user, we send N/10+1 requests for all of the projects to the DB instead of just two. Given that the number of project is only going to increase, I believe this approach has no future.
Remember how we recently had to limit the number of components loaded for display on the main page to improve performance? When we have to do this for projects, it will only make the +1 part faster. N/10 calls for clearing status will still be loading all projects. Same applies to the sticky filtering by Group, which is already being used.
I'm afraid that there is no acceptable way of loading deep project tree structures on the fly. I think we will have to compute this information in some background jobs and load it from the DB cache when necessary.
Case in point: my dev system contains 1099 projects. It took 2:25 min for the projects page to completely load, including clearing statuses.

Let's discuss.

Reply:
Correction: the N/10 part _is_ affected by filtering the initial list of projects as the N gets smaller. I don't think, however, that this invalidates my argument.

Then there is the option of decreasing the number of calls by increasing the denominator, i.e. loading clearing statuses for more projects per request. I tested with 100 projects per request: 1:22 min. That's faster, but not qualitatively better.

Reply:
Hi @alexbrdn - sure, maybe as a start after the call today? And if we want to research in more depth in person, let's do this in two weeks when we meet.

Reply:
As discussed, I used somewhat of a cache to only query all projects once in 2 minutes. It would be good if you could retest with your bigger set of test data.